controllers/vspherevm_controller.go

```diff
@@ -156,13 +157,54 @@ func (r vmReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, reterr error)
 	}
 	conditions.MarkTrue(vsphereVM, infrav1.VCenterAvailableCondition)
 
+	var vsphereFailureDomain *infrav1.VSphereFailureDomain
```

Comment:
what happens when we call `reconcileVM` on an HAProxy VM ? are we checking ctx.VSphereFailureDomain is not nil ?

Reply:
Yes the `reconcileVMGroupInfo` has a nil check before using it.